File: 프로그래머스/0/120875. 평행/평행.swift
import Foundation

func solution(_ dots:[[Int]]) -> Int {
    let p1 = dots[0]
    let p2 = dots[1]
    let p3 = dots[2]
    let p4 = dots[3]
    if (p1[1] - p2[1]) * (p3[0] - p4[0]) == (p3[1] - p4[1]) * (p1[0] - p2[0])  { return 1 }
    if (p1[1] - p3[1]) * (p2[0] - p4[0]) == (p2[1] - p4[1]) * (p1[0] - p3[0]) { return 1 }
    if (p1[1] - p4[1]) * (p2[0] - p3[0]) == (p2[1] - p3[1]) * (p1[0] - p4[0]) { return 1 }
    return 0
}

File: 프로그래머스/0/120875. 평행/평행.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

double getSlope(vector<int> a, vector<int> b) { return double(b[1] - a[1]) / double(b[0] - a[0]); }

int solution(vector<vector<int>> dots) {
    int answer = 0;

    if (getSlope(dots[0], dots[1]) == getSlope(dots[2], dots[3]) || getSlope(dots[0], dots[2]) == getSlope(dots[1], dots[3]) || getSlope(dots[0], dots[3]) == getSlope(dots[1], dots[2])) {
        answer = 1;
    }

    return answer;
}